### ISSUE TITLE
use dropout to avoid overfitting

### ISSUE THREAD

**Issue by benoitfragit** [benoitfragit/libBrain#51]
Only need an xml option

**Comment by benoitfragit** [comment]
For each input and hidden layer before submitted any new input vector in training mode, we update the value of the bernoulli random variable . Each input (except bias) is associated to such a variable. It result for the activation of the neuron 

a = sigma(sum(x_i * w_i * d_i))

Where d_i is the dropout factor associated to the input i

**Comment by benoitfragit** [comment]
See
https://iamtrask.github.io/2015/07/28/dropout/
